core/object/callable_method_pointer.cpp

```diff
@@ -68,16 +68,19 @@ uint32_t CallableCustomMethodPointerBase::hash() const {
 	return h;
 }
 
-void CallableCustomMethodPointerBase::_setup(uint32_t *p_base_ptr, uint32_t p_ptr_size) {
+void CallableCustomMethodPointerBase::_setup(const char *p_base_ptr, uint32_t p_ptr_size) {
 	comp_ptr = p_base_ptr;
-	comp_size = p_ptr_size / 4;
+	comp_size = p_ptr_size;
+	h = 0;
 
 	// Precompute hash.
-	for (uint32_t i = 0; i < comp_size; i++) {
+	for (uint32_t i = 0; i < comp_size; i += sizeof(uint32_t)) {
```

Comment:
This needs either a check that `comp_size % 4 == 0`, or handle the case where it isn't (and somehow hash the remaining bits). Given the legacy code also didn't handle this, it should probably be the former.

Reply:
As-is, this should hit the trailing bytes and hash them.  That's why there's the `MIN` below.

This would result in a hash change for things that weren't a multiple of 4-bytes, but that doesn't seem to be the case anywhere it's currently used.

Reply:
Oh right, I totally missed that. That should be good then :)